lib/pbench/server/auth/__init__.py

```diff
@@ -63,6 +63,7 @@ def __init__(
         self.verify = verify
         self.connection = requests.session()
         self.set_well_known_endpoints()
+        self.set_oidc_public_key()
```

Comment:
On top of Dave and Peter's concerns, I feel impelled to point out that it's a Bad Idea to call instance methods from inside the c'tor.  It's OK to call static member functions and the like (and member functions of _other_ classes, like we do at line 64), but, until the c'tor returns, the instance isn't fully constructed, and so we really shouldn't be invoking methods on it.

I realize that doing otherwise is highly inconvenient, here, but the code as it stands is making an implicit assumption:  the construction is complete as of line 64, and so it is "safe" to now start using the object.  We should make this assumption explicit with a bid ol' hairy block comment which explains that, prior to this line we're still constructing the object, and after this line we're starting to _use_ it, and that any future changes need to be made accordingly.